src/webpack/CompilerEventsFacade.ts

```diff
@@ -5,36 +5,36 @@ export default class CompilerEventsFacade {
 
   private _compiler: Compiler;
 
-  private _legacyTapable: boolean;
-
   constructor(compiler: Compiler) {
     this._compiler = compiler;
-    this._legacyTapable = !compiler.hooks;
   }
 
   public afterOptimizeChunkAssets(
     call: (compilation: Compilation, chunks: Compilation["chunks"]) => void,
   ) {
-    return this._legacyTapable
-      ? (this._compiler as any).plugin("compilation",comp =>
-          comp.plugin("after-optimize-chunk-assets",chunks =>
-            call(comp, chunks),
-          ),
-        )
-      : this._compiler.hooks.compilation.tap(
-          CompilerEventsFacade.extensionName,comp =>
-            comp.hooks.afterOptimizeChunkAssets.tap(
-              CompilerEventsFacade.extensionName,chunks => call(comp, chunks),
-            ),
+    return this._compiler.hooks.compilation.tap(
+      CompilerEventsFacade.extensionName,
+      (comp) => {
+        const afterOptimizeChunkAssets = (chunks) => {
+          call(comp, chunks);
+        };
+        /* https://github.com/webpack/webpack/blob/main/lib/Compilation.js#L772-L779
+        afterOptimizeChunkAssets = PROCESS_ASSETS_STAGE_OPTIMIZE + 1
+        afterOptimizeChunkAssets = 101
+        */
+        const stage = 101;
```

Comment:
How are we getting 101 here? Additionally, we shouldn't hard code this but leave it referenceable.

You can just get it from `Compilation`.

```const stage = Compilation.PROCESS_ASSETS_STAGE_OPTIMIZE + 1```

Reply:
I had just hardcoded the value here for debugging and forgot to reverse the change to make sure I was going about this the right way. Fixed now!